netmiko/nokia/nokia_srl.py

```diff
@@ -0,0 +1,180 @@
+        """Commit changes by using 'commit stay'."""
+        cmd = "commit stay"
+        output = self._send_command_str(
+            command_string=cmd, strip_prompt=False, strip_command=False
+        )
+        return output
+
+    def save_config(
+        self,
+        cmd: str = "save startup",
+        confirm: bool = False,
+        confirm_response: str = "",
+    ) -> str:
+        """Save current running configuration as initial (startup) configuration"""
+        output = self._send_command_str(
+            command_string=cmd, strip_prompt=False, strip_command=False
+        )
+        return output
+
+    def exit_config_mode(self, exit_config: str = "", pattern: str = "") -> str:
+        """Exit the candidate private mode"""
+        output = ""
+        self.write_channel(self.RETURN)
+        prompt = self.read_until_pattern(pattern="#")
+
+        if self._has_uncommitted_changes(prompt):
+            # Changes were made but not committed. Discarding changes
+            output += self._discard()
+        # Switch to 'running' mode
+        output += self._running_mode()
+        return output
+
+    def send_config_set(
+        self,
+        config_commands: Union[str, Sequence[str], TextIO, None] = None,
+        exit_config_mode: bool = False,
+        **kwargs: Any,
+    ) -> str:
+        """Nokia SRL requires you not exit from configuration mode."""
+        return super().send_config_set(
+            config_commands=config_commands, exit_config_mode=exit_config_mode, **kwargs
+        )
+
+    def _discard(self) -> str:
+        """Discard changes made in candidate private mode"""
+        log.warning("Uncommitted changes will be discarted!")
+        cmd = "discard stay"
+        output = self._send_command_str(
+            command_string=cmd, strip_prompt=False, strip_command=False
+        )
+        return output
+
+    def _running_mode(self) -> str:
+        """Enter running mode"""
+        cmd = "enter running"
+        output = self._send_command_str(
+            command_string=cmd, strip_prompt=False, strip_command=False
+        )
+        return output
+
+    def _has_uncommitted_changes(self, prompt: str) -> bool:
+        """
+        The asterix (*) next to the mode name indicates that the candidate configuration
+        has changes that have not yet been committed.
+
+        The plus sign (+) in the prompt indicates that the running configuration differs
+        from the startup configuration. After you enter the save startup command,
+        the running configuration is synchronized with the startup configuration,
+        and the plus sign is removed from the prompt.
+
+        The exclamation mark (!) in the prompt indicates that another user has commited
+        changes to the running datastore.
+        """
+        matches = re.search(r"\n--{( | \* | \+ | \+\* | \!\+ | \!\* )candidate", prompt)
+        return True if matches and "*" in matches.group() else False
```

Comment:
So if you do the `commit stay` that synchronizes candidate to running. Does it also synchronize running to startup or do you separately need to execute `save startup` for that?